src/sparse/impl/KokkosSparse_spgemm_jacobi_denseacc_impl.hpp

```diff
@@ -0,0 +1,301 @@
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include "KokkosKernels_Utils.hpp"
+
+namespace KokkosSparse{
+
+  namespace Impl{
+
+    template <typename HandleType,
+	      typename a_row_view_t_, typename a_lno_nnz_view_t_, typename a_scalar_nnz_view_t_,
+	      typename b_lno_row_view_t_, typename b_lno_nnz_view_t_, typename b_scalar_nnz_view_t_  >
+    template <typename a_row_view_t, typename a_nnz_view_t, typename a_scalar_view_t,
+	      typename b_row_view_t, typename b_nnz_view_t, typename b_scalar_view_t,
+	      typename c_row_view_t, typename c_nnz_view_t, typename c_scalar_view_t,
+	      typename dinv_view_t,
+	      typename mpool_type>
+    struct KokkosSPGEMM
+    <HandleType, a_row_view_t_, a_lno_nnz_view_t_, a_scalar_nnz_view_t_,
+     b_lno_row_view_t_, b_lno_nnz_view_t_, b_scalar_nnz_view_t_>::
+    JacobiSpGEMMDenseAcc
+    {
+      nnz_lno_t numrows;
+      nnz_lno_t numcols;
+
+      a_row_view_t row_mapA;
+      a_nnz_view_t entriesA;
+      a_scalar_view_t valuesA;
+
+      b_row_view_t row_mapB;
+      b_nnz_view_t entriesB;
+      b_scalar_view_t valuesB;
+
+      c_row_view_t row_mapC;
+      c_nnz_view_t entriesC;
+      c_scalar_view_t valuesC;
+
+      typename c_scalar_view_t::const_value_type omega;
+      dinv_view_t dinv;
+
+      nnz_lno_t *pEntriesC;
+      scalar_t *pVals;
+
+      mpool_type memory_space;
+      const KokkosKernels::Impl::ExecSpaceType my_exec_space;
+      const nnz_lno_t team_work_size;
+
+
+      JacobiSpGEMMDenseAcc(nnz_lno_t m_, nnz_lno_t k_,
+			   a_row_view_t row_mapA_, a_nnz_view_t entriesA_, a_scalar_view_t valuesA_,
+			   b_row_view_t row_mapB_, b_nnz_view_t entriesB_, b_scalar_view_t valuesB_,
+			   c_row_view_t row_mapC_, c_nnz_view_t entriesC_, c_scalar_view_t valuesC_,
+			   typename c_scalar_view_t::const_value_type omega_, dinv_view_t dinv_,
+			   mpool_type memory_space_,
+			   const KokkosKernels::Impl::ExecSpaceType my_exec_space_,
+			   nnz_lno_t team_row_chunk_size):
+	numrows(m_), numcols(k_),
+        row_mapA(row_mapA_), entriesA(entriesA_), valuesA(valuesA_),
+        row_mapB(row_mapB_), entriesB(entriesB_), valuesB(valuesB_),
+        row_mapC(row_mapC_), entriesC(entriesC_), valuesC(valuesC_),
+	omega(omega_), dinv(dinv_),
+	pEntriesC(entriesC_.data()), pVals(valuesC.data()),
+        memory_space(memory_space_),
+        my_exec_space(my_exec_space_),
+        team_work_size(team_row_chunk_size){
+      }
+
+      KOKKOS_INLINE_FUNCTION
+      size_t get_thread_id(const size_t row_index) const{
+	switch (my_exec_space){
+	default:
+	  return row_index;
+#if defined( KOKKOS_ENABLE_SERIAL )
+	case KokkosKernels::Impl::Exec_SERIAL:
+	  return 0;
+#endif
+#if defined( KOKKOS_ENABLE_OPENMP )
+	case KokkosKernels::Impl::Exec_OMP:
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE
```

Comment:
It is better if our new codes shouldn't. depend on deprecated code, can we remove that ?

Reply:
Done.